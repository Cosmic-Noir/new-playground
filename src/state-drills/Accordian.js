import React from "react";

class Accordian extends React.Component {
  static defaultProps = {
    tabs: []
  };

  render() {
    return <div />;
  }
}

export default Accordian;
