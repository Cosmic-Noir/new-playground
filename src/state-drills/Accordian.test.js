import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Accordian from "./Accordian";

describe(`Accordian component`, () => {
  const sections = [
    {
      title: "Section 1",
      content: "Lorem ipsum dolor sit amet consectetur adipisicing elit."
    },
    {
      title: "Section 2",
      content:
        "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!"
    },
    {
      title: "Section 3",
      content:
        "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?"
    }
  ];

  // Smoke screen test without input
  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Snapshot with no input:
  it("renders empty given no input", () => {
    const wrapper = shallow(<Accordian />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  //Snapshot rendering content by default
  it("renders the first content section", () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // Snapshot rendering content of second button after clicked
  it("renders the second listItems content", () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
