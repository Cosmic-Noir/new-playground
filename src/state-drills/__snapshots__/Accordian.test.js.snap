// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian component renders empty given no input 1`] = `<ul />`;

exports[`Accordian component renders the first content section 1`] = `
<ul>
  <button
    key="0"
    onClick={[Function]}
  >
    Section 1
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Section 2
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Section 3
  </button>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit.
  </p>
</ul>
`;

exports[`Accordian component renders the second listItems content 1`] = `
<ul>
  <button
    key="0"
    onClick={[Function]}
  >
    Section 1
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Section 2
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Section 3
  </button>
  <p>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </p>
</ul>
`;
